tests/MessagePack.SourceGenerator.Tests/MessagePack.SourceGenerator.Tests.csproj

```diff
@@ -24,7 +24,10 @@
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\..\src\MessagePack.SourceGenerator\MessagePack.SourceGenerator.csproj" />
+    <ProjectReference Include="..\..\src\MessagePack.SourceGenerator\MessagePack.SourceGenerator.csproj">
+      <!-- https://github.com/dotnet/roslyn-sdk/issues/850 -->
+      <OutputItemType>Analyzer</OutputItemType>
```

Comment:
In my experience (and I've written several analyzers and a few source generators), this metadata will cause the source generator to run on this project. But this project isn't supposed to run this source generator as part of its own build. This project _tests_ the source generator via its APIs. You can debug the analyzer simply by debugging the tests in this project. 
The test project that exercises the source generator in _this_ way that you're going for here is `MessagePack.SourceGenerator.ExecutionTests`, and it already applies this metadata to the ProjectReference item.

So you see, we have two testing strategies, one of which is what you seem to be going for here. We _don't want_ the API-based testing project to itself consume the analyzer/generator as part of its own _build_. 

And as you don't actually add any `[MessagePackObject]` types to this test project, I wonder what you were hoping to accomplish by adding this metadata, since the analyzer wouldn't have anything to do during the build of this project anyway.